tests/test_slips_utils.py

```diff
@@ -1,10 +1,307 @@
+    if side_effect:
+        with patch('requests.get', side_effect=side_effect):
+            assert expected_result(utils), "Should handle the situation gracefully"
+    else:
+        assert expected_result(utils), "Should return a list of IPs"
+
+
+@pytest.mark.parametrize("port", [80, 65535, 0])
+def test_is_port_in_use(port):
+    utils = ModuleFactory().create_utils_obj()
+    assert isinstance(utils.is_port_in_use(port), bool)
+
+    if port != 0:
+        assert utils.is_port_in_use(port) is False, f"The port {port} should not be identified as in use."
+    else:
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as temp_socket:
+            temp_socket.bind(('localhost', 0))
+            temp_socket.listen(1)
+            used_port = temp_socket.getsockname()[1]
+            assert utils.is_port_in_use(used_port) is True, "The port should be identified as in use."
+
+
+def test_is_valid_threat_level():
+    utils = ModuleFactory().create_utils_obj()
+    assert utils.is_valid_threat_level('info')
+    assert utils.is_valid_threat_level('low')
+    assert utils.is_valid_threat_level('medium')
+    assert utils.is_valid_threat_level('high')
+    assert utils.is_valid_threat_level('critical')
+    assert not utils.is_valid_threat_level('undefined')
+
+
+def test_convert_to_mb():
+    utils = ModuleFactory().create_utils_obj()
+    assert utils.convert_to_mb(1000000) == 1
+    assert utils.convert_to_mb(5000000) == 5
+    assert utils.convert_to_mb(10 ** 12) == 10 ** 6
+    assert utils.convert_to_mb(1048576) == 1.048576
+    assert utils.convert_to_mb(0) == 0
+
+
+def test_is_msg_intended_for():
+    utils = ModuleFactory().create_utils_obj()
+    message = {"data": "Some data", "channel": "test_channel"}
+    assert utils.is_msg_intended_for(message, "test_channel") is True
+
+    message = {"data": "Some data", "channel": "other_channel"}
+    assert utils.is_msg_intended_for(message, "test_channel") is False
+
+    message = {"data": 123, "channel": "test_channel"}
+    assert utils.is_msg_intended_for(message, "test_channel") is False
+
+
+def test_convert_format_between_non_default_formats():
+    utils = ModuleFactory().create_utils_obj()
+    original_format = '2023-04-01 12:00:00'
+    expected_result = '04/01/2023 12:00 PM'
+    assert utils.convert_format(original_format, '%m/%d/%Y %I:%M %p') == expected_result
+
+
+@dataclass
+class MockDataClass:
+    id: int
+    name: str
+
+
+class MockEnum(Enum):
+    TYPE1 = "Type 1"
+    TYPE2 = "Type 2"
+
+
+@pytest.mark.parametrize("input_value, expected_output", [
+    (MockDataClass(id=1, name="Test"), {"id": 1, "name": "Test"}),
+    (MockEnum.TYPE1, "Type 1"),
+    ({"key1": MockDataClass(id=2, name="Nested"), "key2": [MockEnum.TYPE1, MockEnum.TYPE2]},
+     {"key1": {"id": 2, "name": "Nested"}, "key2": ["Type 1", "Type 2"]})
+])
+def test_to_json_serializable(input_value, expected_output):
+    utils = ModuleFactory().create_utils_obj()
```

Comment:
i believe you forgot to commit the logic here too, usually pre-commit hooks catch errors like these, so you'll be able to easily detect them in the future